docs/guides/advanced/data/4844_data.md

```diff
@@ -0,0 +1,185 @@
+
+**Why did Ethereum introduce these commitments instead of continuing with the previous hash-based system?**
+
+The short answer is, that while the current version of Ethereum doesn't gain much from this change, the next version,
+known as 'full Danksharding,' will greatly benefit from these commitments, as they will allow the addition of many more
+blobs.
+
+## Technical details
+
+Let's go back to our problem statement:
+
+**We have 128kb of data in L2, and we want to prove on L1 that it was correctly 'attached' to L1 (that is - that it was
+made available).**
+
+This verification is done in 14 steps, divided into three parts: inside the prover, inside the state keeper, and finally
+within the L1 contract.
+
+### Inside the prover
+
+Most of this is done in either witness generator or a new circuit type 4844:
+
+| step                       | explanation                                                                                                                                                                                                                                  |
+| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Split data                 | Divide the data into 4096 units of 256-bit points.                                                                                                                                                                                           |
+| Create Polynomial          | Form a polynomial `p` that passes through these points. For more clarity on this, refer to [Blobspace 101][blobspace_101]                                                                                                                    |
+| Trusted setup              | Obtain the [trusted setup][trusted_setup] from Ethereum's KZG ceremony (also known as the secret point 's').                                                                                                                                 |
+| Compute Commitment         | [Calculate the commitment][compute_commitment] `(c = p(s))`.                                                                                                                                                                                 |
+| Versioned Hash:            | Generate a versioned hash (`vh = hash(c)`) which is more flexible and smaller in size than the commitment. (32 bytes vs 48)                                                                                                                  |
+| Choose a Random Point:     | Select a 'random' point `x = hash(hash(data) \|\| vh)` using the [hash of the data and the versioned hash][random_point_selection]. This is ['Fiat-Shamir' heuristic][fiat_shamir]                                                           |
```

Comment:
below we write that x is not generated using hash(data)...

heuristic --> transform